src/lazy/encoder/binary/v1_1/value_writer.rs

```diff
@@ -971,10 +974,20 @@ mod tests {
             f64::INFINITY,
             f64::NEG_INFINITY,
             f64::NAN,
+            f64::PI(),
+            f64::E(),
+            f64::EPSILON,
         ];
         for value in test_f64s {
-            let mut expected_encoding = vec![0x5D];
-            expected_encoding.extend_from_slice(&value.to_le_bytes()[..]);
+            let mut expected_encoding = vec![];
+            let float32 = *value as f32;
+            if float32 as f64 == *value {
+                expected_encoding.push(0x5C);
+                expected_encoding.extend_from_slice(&float32.to_le_bytes()[..]);
+            } else {
+                expected_encoding.push(0x5D);
+                expected_encoding.extend_from_slice(&value.to_le_bytes()[..]);
+            }
```

Comment:
🗺️ We have a lot of unit tests where we re-invent the wheel for the sake of bootstrapping the encoder/decoder. We should look at migrating some of the more painful ones (like this one) to a roundtrip test using our recently established readers and writers.